src/eradiate/radprops/_array.py

```diff
@@ -0,0 +1,254 @@
+"""
+Array radiative profile.
+"""
+
+from __future__ import annotations
+
+import typing as t
+
+import attrs
+import numpy as np
+import pint
+import xarray as xr
+
+from ._core import RadProfile, ZGrid, make_dataset
+from ..attrs import define, documented
+from ..units import to_quantity
+from ..units import unit_registry as ureg
+
+
+@define(eq=False)
+class ArrayRadProfile(RadProfile):
+    """
+    Array radiative profile.
+
+    This class provides an interface to generate vertical profiles of
+    atmospheric volume radiative properties (also sometimes referred to as
+    collision coefficients).
+
+    The array radiative profile is built from absorption and scattering
+    coefficient data provided as input. This can be useful for debugging and
+    benchmarking.
+    """
+
+    sigma_a: xr.DataArray | None = documented(
+        attrs.field(
+            default=None,
+        ),
+        doc="``DataArray`` of absorption coefficients "
+        "The ``DataArray`` is composed of two dimensions ``{w, z}`` representing the "
+        "wavelength and altitude respectively. Note that ``w`` must be of length "
+        "2 minimum to be correctly interpolated. ",
+        type="DataArray or None",
+        init_type="DataArray or None",
+        default="None",
+    )
+
+    sigma_s: xr.DataArray | None = documented(
+        attrs.field(
+            default=None,
+        ),
+        doc="``DataArray`` of scattering coefficients "
+        "The ``DataArray`` is composed of two dimensions ``{w, z}`` representing the "
+        "wavelength and altitude respectively. Note that ``w`` must be of length "
+        "2 minimum to be correctly interpolated. ",
+        type="DataArray or None",
+        init_type="DataArray or None",
+        default="None",
+    )
+
+    has_absorption: bool = documented(
+        attrs.field(
+            default=True,
+            converter=bool,
+            validator=attrs.validators.instance_of(bool),
+        ),
+        doc="Absorption switch. If ``True``, the absorption coefficient is "
+        "computed. Else, the absorption coefficient is not computed and "
+        "instead set to zero.",
+        type="bool",
+        default="True",
+    )
+
+    has_scattering: bool = documented(
+        attrs.field(
+            default=True,
+            converter=bool,
+            validator=attrs.validators.instance_of(bool),
+        ),
+        doc="Scattering switch. If ``True``, the scattering coefficient is "
+        "computed. Else, the scattering coefficient is not computed and "
+        "instead set to zero.",
+        type="bool",
+        default="True",
+    )
+
+    rayleigh_depolarization: np.ndarray = documented(
+        attrs.field(
+            converter=lambda x: np.array(x, dtype=np.float64),
+            kw_only=True,
+            factory=lambda: np.array(0.0),
+        ),
+        type="ndarray",
```

Comment:
I missed this previously: specifying the default and the allowed init types would be good (see also on the `AtmosphereRadProfile` side).